core/modules/qproc/QuerySession.h

```diff
@@ -215,6 +217,10 @@ class QuerySession {
     ChunkSpecVector _chunks; ///< Chunk coverage
     std::shared_ptr<QueryPluginPtrVector> _plugins; ///< Analysis plugin chain
 
+    /// Maximum number of chunks in an interactive query. TODO: Maybe put in config file.
```

Comment:
I think in the long tern the decision should be based on query cost estimate (if we ever get that done) which by itself could be configured from config file. Maybe it's worth to add rudimentary "cost estimate" thing and its configuration (in a new ticket).